contracts/contracts/core/ProviderRegistry.sol

```diff
@@ -267,20 +268,20 @@ contract ProviderRegistry is
 
     /// @dev Ensure the provider's balance is greater than minStake and no pending withdrawal
     function isProviderValid(address provider) public view {
-        require(providerStakes[provider] >= minStake, "Insufficient stake");
-        require(withdrawalRequests[provider] == 0, "Pending withdrawal request");
+        require(providerStakes[provider] >= minStake, InsufficientStake());
```

Comment:
I think this is a general comment for all of these errors. I think we should add more arguments here that will help identify the problems. For example the provider address here.